ush/syndat_qctropcy.sh

```diff
@@ -124,166 +118,164 @@ set_trace
+    echo 1900010100 > dateck
+    set +x
+    echo -e "\n${msg}\n"
+    set_trace
 fi
 
-
 #  Generate the correct RUNID and FILES value based on $NET, $RUN and $cyc
 #  Copy this into parm file, then cat the remaining switches in the parm file
 #     Note: FILES=T for 00Z GDAS at tm00 (last run of day centered on 00Z)
 #                       06Z GDAS at tm00 (last run of day centered on 06Z)
 #                       12Z GDAS at tm00 (last run of day centered on 12Z)
 #                       18Z GDAS at tm00 (last run of day centered on 18Z)
 
-net=$NET
+net="${NET}"
 files=F,
-if [ "$RUN" = 'ndas' ]; then
-   net=ndas
-elif [ "$RUN" = 'gdas' ]; then
-   files=T,
+if [[ "${RUN}" == 'ndas' ]]; then
+    net=ndas
+elif [[ "${RUN}" == 'gdas' ]]; then
+    files=T,
 fi
-if [ -n "$files_override" ]; then  # for testing, typically want FILES=F
-  files_override=$(echo "$files_override" | tr [a-z] [A-Z] | tr -d [.] | cut -c 1)
-  if [ "$files_override" = 'T' -o "$files_override" = 'F' ]; then
-    msg="***WARNING: Variable files setting will be overriden from $files to $files_override. Override expected if testing."
-    files=$files_override
-  else
-    msg="***WARNING: Invalid attempt to override files setting. Will stay with default for this job"
-  fi
-  set +x
-  echo -e "\n${msg}\n"
-  set_trace
-  echo $msg >> $pgmout
+
+if [[ -n "${files_override}" ]]; then # for testing, typically want FILES=F
+    files_override=${files_override^^}
+    files_override=${files_override//./}
+    files_override=${files_override:0:1}
+    if [[ "${files_override}" == 'T' || "${files_override}" == 'F' ]]; then
+        msg="WARNING: Variable files setting will be overriden from ${files} to ${files_override}. Override expected if testing."
+        files=${files_override}
+    else
+        msg="WARNING: Invalid attempt to override files setting. Will stay with default for this job"
+    fi
+    set +x
+    echo -e "\n${msg}\n"
+    set_trace
 fi
 
-echo " &INPUT  RUNID = '${net}_${tmmark}_${cyc}', FILES = $files " > vitchk.inp
-cat ${PARMgfs}/relo/syndat_qctropcy.${RUN}.parm >> vitchk.inp
+echo " &INPUT  RUNID = '${net}_${tmmark}_${cyc}', FILES = ${files} " > vitchk.inp
+cat "${PARMgfs}/relo/syndat_qctropcy.${RUN}.parm" >> vitchk.inp
 
 #  Copy the fixed fields
 
-cpreq ${FIXgfs}/am/syndat_fildef.vit fildef.vit
-cpreq ${FIXgfs}/am/syndat_stmnames stmnames
-
+cpreq "${FIXgfs}/am/syndat_fildef.vit" fildef.vit
+cpreq "${FIXgfs}/am/syndat_stmnames" stmnames
 
 rm -f nhc fnoc lthistry
 
-
 #########################################################################
 
 #  There are five possible sources of tropical cyclone bogus messages
 #  All are input to program syndat_qctropcy
 #  ------------------------------------------------------------------
 
-if [ -s ${HOMENHC}/tcvitals ]; then
-   echo "tcvitals found" >> $pgmout
-   cpreq ${HOMENHC}/tcvitals nhc
+if [[ -s "${HOMENHC}/tcvitals" ]]; then
+    echo "tcvitals found"
+    cp "${HOMENHC}/tcvitals" nhc
```

Comment:
I think this should still be `cpreq`
```suggestion
    cpreq "${HOMENHC}/tcvitals" nhc
```

Reply:
Fixed